app/views/stories/tagged_articles/_meta.html.erb

```diff
@@ -1,4 +1,4 @@
-<% display_tag_name = @tag.name.capitalize %>
+<% display_tag_name = @tag.pretty_name.present? ? @tag.pretty_name : @tag.name.capitalize %>
```

Comment:
I see - you are covering for the case where a tag pretty name existed, and the tag moderator/admin removed it, and an empty string (rather than nil) was saved. Makes sense.

Since this variable is also for the meta (what gets included in meta tags like `twitter:description`, `og:description` or `og:title` meta properties), does it make sense to modify only the title_with_timeframe (the window or tab title) tag on the next line rather than this variable?

A good example for other reviewers is https://dev.to/t/cpp/latest - the pretty name is "C++", the title is "Cpp - Latest posts - DEV Community", under this change the title would be "C++ - Latest posts - DEV Community"

![cpp tag latest posts page](https://user-images.githubusercontent.com/1237369/152247301-0042158f-8985-440a-8281-827017efe9f8.png)

I think it would be safer to adapt just the use of display tag name on line 3 immediately after this (leaving the meta tags the same). Especially for cases like the twitter creator meta later (where the tag's identifier, rather than it's pretty name, is used like a user handle)?  

```html
<meta name="twitter:creator" content="@<%= display_tag_name %>">
```

Reply:
We have specs in place that are validating the title (but aren't setting the pretty name) - https://github.com/forem/forem/blob/main/spec/requests/stories/tagged_articles_spec.rb#L228-L231 and https://github.com/forem/forem/blob/main/spec/requests/stories/tagged_articles_spec.rb#L241-247, right now those are only seeing tags with no pretty name (so passes by falling back to the original tag.name behavior).

If I update the let block for that context on line 179 to add a pretty name:

```ruby
          let(:tag) { create(:tag, pretty_name: Faker::Lorem.sentence) }
```

Then the tests for title (page 1 and page 2) start to fail. I think it makes sense to test both cases so we know this behaves as expected.

If the page 1 tag (declared on line 179's let block) had a pretty name then the page one title would use pretty_name, and if the page 2 test `it "renders proper page 2"` replaced that with a default tag it covers both sides of the conditional?

```ruby
     context "without user signed in" do
-          let(:tag) { create(:tag) }
+          let(:tag) { create(:tag, pretty_name: Faker::Lorem.sentence) }

           def renders_title(tag)
-            expect(response.body).to include("<title>#{tag.name.capitalize} - ")
+            expect(response.body).to include("<title>#{tag.pretty_name} - ")
           end

           it "renders proper page 2", :aggregate_failures do
+            tag = create(:tag)
             create_list(:article, 20, user: user, featured: true, tags: [tag.name], score: 20)
             get "/t/#{tag.name}/page/2"
```